server/backend.py

```diff
@@ -39,15 +41,34 @@ async def main():
 
 				try:
 					client = nasc.NASCClient()
-					client.set_title(0x0004013000003202, 20)
-					client.set_device(SERIAL_NUMBER, MAC_ADDRESS, DEVICE_CERT, DEVICE_NAME)
+					client.set_title(0x0004013000003202, 20) # to be honest, this should be seperate between networks, so if one eg, gets banned, you still get to keep the friend code for the other network, but i'm lazy.
 					client.set_locale(REGION, LANGUAGE)
-					client.set_user(PID, PID_HMAC)
 
+					if network == 0: # Nintendo Network
+						client.set_url("nasc.nintendowifi.net")
+						PID = NINTENDO_PID
+						NEX_PASSWORD = NINTENDO_NEX_PASSWORD
+						
+					elif network == 1:
+						client.set_url("nasc.pretendo.cc")
+						client.context.set_authority(None)
+						PID = PRETENDO_PID
+						NEX_PASSWORD = PRETENDO_NEX_PASSWORD
+						
+					else:
+						raise Exception(NetworkIDsToName(network).name + " is not a valid network")
+					
+					client.set_device(SERIAL_NUMBER, MAC_ADDRESS, DEVICE_CERT, DEVICE_NAME)
+					client.set_user(PID , PID_HMAC)
+					
 					response = await client.login(0x3200)
 
 					s = settings.load('friends')
 					s.configure("ridfebb9", 20000)
+
+					if network == 1: # If the app starts randomly hanging on Pretendo, try removing this, and the next line.
+						s["prudp.ping_timeout"] = 100000000000 # oh my god this is horrifying, but it makes it works, so who am i to care
```

Comment:
We made a lot of stability changes and bug fixes a while ago, after this PR was made. This hack can likely be removed, and if the issue still persists then you should open an issue about it on our end. This is obviously bad behavior on our end if it persists

Reply:
Hi there! A bit ago, I tested, and the program does work without these lines. (yay, thank you!) However, the PR has technically been merged (see https://github.com/MCMi460/3DS-RPC/commit/8751376416cdecdc93697052fa157fedec3c6a31), so I can't modify the PR without causing problems (probably). Why this isn't closed, I do not know. I should probably close it. I'll make sure to try to get the owner of the repo to remove the workaround. Thanks tho!